source/tomopy/util/__init__.py

```diff
@@ -48,3 +48,38 @@
 
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
+
+import ctypes
+import os
+import sys
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+def c_shared_lib(lib_name, do_warn=True):
+    """Get the path and import the C-shared library."""
+    load_dll = ctypes.cdll.LoadLibrary
+    ext = '.so'
+    if sys.platform == 'darwin':
+        ext = '.dylib'
+    if os.name == 'nt':
+        ext = '.dll'
+        load_dll = ctypes.windll.LoadLibrary
+    base_path = os.path.abspath(os.path.dirname(
+                                os.path.dirname(__file__)))
+    sharedlib = os.path.join(base_path, 'sharedlibs', '%s%s' % (lib_name, ext))
+    if os.path.exists(sharedlib):
+        return load_dll(sharedlib)
+    # cannot find shared lib:
+    if do_warn is True:
+        logger.warning('OSError: ' +
+                       'The following shared lib is missing!\n{}'.format(
+                           sharedlib))
+
+
+def MissingLibrary(function):
+    print(f"The {function} algorithm is unavailable."
+          " Check CMake logs to determine if TomoPy was"
+          " built with dependencies required by this algorithm.")
```

Comment:
Why are these defined in `util.__init__` instead of `extern.__init__`? Do other modules in `util` use these functions? If not, please move these to into the `extern` module.

Reply:
No other modules in util use the function. However, if they are placed in extern.__init__, it causes a problem of circular dependencies because extern.__init__ is importing accel, gridrec, prep, misc, and recon, but those same files would have to then import extern.__init__ to access the above functions. I've tried changing the order of import to no success. The only solution I've found is to place the above functions in util.__init__, which gets rid of the circular dependency.

Reply:
Which version of Python are you using?

Reply:
Python 3.7.6.

Reply:
Python 3.7.6.

Reply:
Works for me on python 3.7.6 and 3.6.10.

Reply:
I see. I think my issue was that I wasn't telling the files the correct place to get the c_shared_lib function.